### DISCUSSION TITLE
Icon usage in a free software (licensing)

### DISCUSSION THREAD

**Discussion by zkvvoob** [elrumo/macOS_Big_Sur_icons_replacements#1382] [General]
Hello,

I'm trying to convince the developer(s) of Calibre (a free ebook management software) to finally switch to an icon that is consistent with the macOS Big Sur design. I've suggested the icons available on this website, but they are hesitant due to the lack of clarity on the icons' usage rights/license.

Could you please shead some light on the matter, please?

Thanks!

**Comment by elsiehupp** [comment]
I vaguely remember running into the same problem myself with the Nextcloud icon, so I ended up [just building it myself](https://github.com/nextcloud/desktop/pull/4631) based on the same "squircle" outline and the existing Nextcloud icon assets. If you want me to make one similar to a particular example from the website that could have clearer licensing, I'd be happy to help.

(The only real issue is that Kovid told me to piss off like six months ago, but he's apparently a grouch to everyone, so he's probably forgotten about the incident entirely. Regardless I could let you do the pull request if you think it would work better.)

**Reply by zkvvoob** [reply]
Thank you very much for agreeing to help, @elsiehupp ! 

I've been chasing Kovid for the past year and a half and he and his community over at mobileread.com have been brushing me off. It was only a few days ago that someone relented that I should suggest an icon and they would see if they could add it to Calibre.

At any rate, if you could spare some time to do an icon of your own that you coud "donate" to Calibre, I think a lot of people would be very grateful!

Personally, I like this one the best: https://media.macosicons.com/parse/files/macOSicons/2ed9758cbc25fc99a3d64f8a1193bb0d_Calibre.icns

**Reply by elrumo** [reply]
@zkvvoob  Apologies if it's too late, but if you want I can get in touch with the one that designed the icon and connect you two. Is the icon the one designed by [ways196](https://macosicons.com/#/u/ways196)?

**Reply by zkvvoob** [reply]
@elrumo, no problem, it's not like the Calibre developer is in any rush. 🤣 
Indeed, the icon I like is designed by ways196.

**Reply by elsiehupp** [reply]
@elrumo personally I like [the very similar one](https://media.macosicons.com/parse/files/macOSicons/b1fde50cf9f27cf434ac172119fafafa_Calibre_Remastered.icns) by [SuperTurtle](https://macosicons.com/#/u/SuperTurtle). Could you try and get in touch with them, as well? Thanks!

**Reply by elsiehupp** [reply]
To explain the reason why I prefer SuperTurtle's version, first, here they are side by side:

<img width="256" alt="Ways196" src="https://user-images.githubusercontent.com/9206310/179604343-3e1a2d10-33da-42c3-9b53-7a12e2bf8676.png"> <img width="256" alt="SuperTurtle" src="https://user-images.githubusercontent.com/9206310/179604360-0a50f385-0723-4c31-943e-6a8036f84b47.png">

The subtle changes in the books don't really make any difference to me, but the apparently slightly thicker corners on Ways196's version I find distracting. If I had to guess, Ways196 scaled down the "squircle" shape, while SuperTurtle used a constant offset.

If you can't get in touch with either of them, I could continue trying to recreate a variation of these two. I just haven't put much time into it either way.

**Comment by elsiehupp** [comment]
One last thing, @elrumo—if you could add explicit license tags to the website, that could super helpful in instances like this. Like basically one can assume that any submitted icons are allowed for personal use, but they probably wouldn't be allowed to be shipped by the application developer themselves. For commercial applications obviously it would make sense for icon designers to want to be paid for their work, but the situation with libre software, especially community-maintained libre software might be a bit different, so at the very least it would be nice for there to be a way for icon designers to tag their work as, e.g., GPL, etc., when submitting icons. (You could also mass-email previous uploaders about the option to specify a license on previous uploads, but that might be a bit spammy.)

(As a separate matter, it could also be nice if icons with permissive or libre licenses could have the option of providing a copy in a more editable format than just the final `.icns`, i.e. Illustrator or Sketch or whatever.)

**Reply by elrumo** [reply]
I've written an initial draft for the licensing, which I will add to the page. What do you think? Is it clear enough, anything that I may be missing?

## Licensing
**All the work found in macOSicons.com can be used free of charge as long as it’s for personal use only.**

**If you’d like to use an icon for anything other than for personal use, from an open-source project, a for-profit business or a project that is shared publicly, you must get permission from me, Elias Ruiz Monserrat, and the icon’s designer.** 

**Generally speaking, there should be no issues with an icon being used in an open-source project, but out of respect for the designer, you should get their permission. In the past, they’ve always been thrilled to help an open-source project. You can get in touch with me, and I’ll try my best to let the designer know that you are interested.**

## Attribution
**If you end up using an icon found in macOSicons.com in your project, you must give credit to the designer. Mentioning macOSicons.com is not necessary, but would be very much appreciated.**

**Reply by elrumo** [reply]
This is in the short term, in the longer term I will look at adding feature you mentioned, where each user can choose a specific licence. The issue could be when the icon is a re-work from an existing commercial licence. I'll need to do more research about that.

**Reply by zkvvoob** [reply]
Sounds reasonable and perfectly straight-forward to me. Thank you!

**Reply by elsiehupp** [reply]
## Thoughts

That looks good for the most part, though I have a few pieces of feedback (in no particular order lol). (I'm using excessive Markdown formatting because of the length of this reply.)

### Implied License

If you feel comfortable doing so on behalf of the designers of past submittals, you might consider using the [CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license (which is essentially a personal-use license), on the basis that Creative Commons licenses have over two decades' worth of real-world legal exposure and have been updated to cover the weird edge cases (hence "4.0" in the instance of CC BY-NC-SA). (Because I assume there was no explicit license granted up until this point, it would be worth explicitly saying that, for icons without an explicit license, the license is an _implied_ personal-use license, with a link to the CC BY-NC-SA 4.0 as an explanation of _a reasonable interpretation_ of the phrase "personal use".)

## Disclaimers of Warranty

Most "free" or "permissive" licenses include a disclaimer of warranty. For example, here's the one from the CC BY-NC-SA 4.0:
  
> No warranties are given. The license may not give you all of the permissions necessary for your intended use. For example, other rights such as publicity, privacy, or moral rights† may limit how you use the material.
> 
> †You may need to get additional permissions before using the material as you intend. [More info](https://wiki.creativecommons.org/wiki/Considerations_for_licensors_and_licensees).

And here's the version from [the GPLv3](https://www.gnu.org/licenses/gpl-3.0.en.html):

> IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR CONVEYS THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.

### Fair Use

Regarding ("Fair Use" [under US trademark law](https://en.wikipedia.org/wiki/Fair_use_(U.S._trademark_law)) and [under US copyright law](https://en.wikipedia.org/wiki/Fair_use)), the Creative Commons Wiki [recommends](https://wiki.creativecommons.org/wiki/Considerations_for_licensors_and_licensees) the following:

> **Indicate rights not covered by the license.**
> 
> Prominently [mark or indicate in a notice any rights held by third parties](https://wiki.creativecommons.org/wiki/Marking/Creators/Marking_third_party_content), such as publicity or trademark rights. This includes any content you used under exceptions or limitations to copyright, and any third party content used under another license (even if it is the same CC license as you applied).

### UI Implementation
  
Basically my interpretation of this is that each icon should be marked with something like:

> \[icon name\] and its related logos may be protected trademarks and are used here under [Fair Use](https://en.wikipedia.org/wiki/Fair_use_(U.S._trademark_law)).

(after a short text like "This icon is provided under an implied CC BY-NC-SA license", with "implied CC BY-NC-SA license" being a link to a page on your site explaining the circumstances for the lack of an explicit license). So for example, an extremely abbreviated version of the text under each icon might read as follows:

> **Appity AppName**
> **Author:** Foobar McFoobar on 15 August 1969
> **Copyright:** presumed CC BY-NC-SA
> **Trademarks:** [Fair Use](https://en.wikipedia.org/wiki/Fair_use_(U.S._trademark_law)).
> 1461

(with the links you already use, as well as an internal link for the implied license and external links for explicit licenses).

### Gatekeeping

In the text "you must get permission from me, Elias Ruiz Monserrat, and the icon’s designer", the "from me, Elias Ruiz Monserrat" part is almost certainly unenforceable, though for icons without explicit licenses you would still be the obvious intermediary for contacting the original authors.

### Future Choice of Licenses

Regarding the choice of licenses you plan on adding to the website for designers submitting icons, my understanding is that Creative Commons licenses are typically considered the more appropriate option for audio and visual media, as well as natural-language text (i.e. not software), though the main exception to this is visual assets accompanying software sometimes being licensed under the same license as the software itself†. Because you would personally control the license-selection interface (as the maintainer of the repository), you could exclude, for example, the "No-Derivative" versions of the Creative Commons license, specifically in order to encourage further remixing.

> †This actually reminded me: regarding @zkvvoob's initial request: [Calibre](https://github.com/kovidgoyal/calibre) is licensed [under the GPLv3](https://github.com/kovidgoyal/calibre/blob/master/LICENSE), ***and this includes [the SVG file used for Calibre's existing icon](https://github.com/kovidgoyal/calibre/blob/master/imgsrc/calibre.svg)***. Because of the nature of the GPL—"You may not propagate or modify a covered work except as expressly provided under this License."—any icons derived from the official Calibre icon would, by virtue of having been provided for distribution, be also mandatorily licensed under the GPL. (It would definitely be worth mentioning this fact on the page explaining the ambiguous nature of the licenses for the existing icon library.)

### Regarding Web Forms

As far as interface-implementation goes, GitHub features [the ability to use templates for Issues and Pull Requests](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates) (which is basically a YAML definition for a web form), and using GitHub as the interface for submissions could make it easier for users to get in touch with icon designers regarding uses beyond those allowed under their chosen license.

The main drawback with using GitHub for icon submission is that graphic designers are less likely to have a GitHub account than software engineers, and it's somewhat non-trivial to learn how to use Git for pull requests (though submitting icons via Issues would be easier for non-technical users to figure out). It is possible, though, that someone has created a GitHub app for creating Pull Requests without using Git (i.e. beyond simple inline patches), but I have not at all looked into it.

Obviously, if you feel more comfortable building your own web forms rather than integrating as much of the pipeline as possible into GitHub, that's your choice, and it's perfectly fine. I suggest GitHub's tools largely because they might require less work to implement, and because of the other advantages offered by GitHub integration, considering your repository is already set up here.

### Developer Certificate of Origin

Another thing worth mentioning with license declarations is that there is another legal document called the [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin), and there is [a GitHub app](https://probot.github.io/apps/dco/) for requiring it on Pull Requests. The main issue with using it is the Fair Use aspect of many of the icons, as well as the ambiguous licensing of the existing icon library, which would make anything derived from the existing icons ineligible for the DCO. Regardless, it's a useful thing to know about!

## Conclusion

Anyway, can you tell I'm kind of a copyright wonk 🤓? If you have any questions or further thoughts about my commentary here, feel free to continue this thread!

**Reply by elsiehupp** [reply]
As another total aside, I noticed when formatting the sample UI implementation that at least for me the website uses US-style date notation. In my sample, I instead used notation with the month spelled out—i.e. "15 August 1969"—in order to reduce the ambiguity between `MM/DD/YYYY` and `DD/MM/YYYY`. I say "at least for me" because for all I know the website uses localization for date notation. (Though if you use localization it could also be worth eventually providing localized versions of the entire website if you felt like figuring that out.)

**Reply by elsiehupp** [reply]
Also, regarding the labor that would go into any of these things, it could be easier to get other people to help you with it if the source for the website itself were explicitly under a "free" or "permissive" license. Like among other things there are specialized services for crowdsourced localization that have GitHub integrations, and you could include a link on the website for people to help with translations.

**Comment by zkvvoob** [comment]
Hello,

I lost track of what happened with our discussion. Have you decided on an official license for the icons? Or have you had a chance to speak to Kovid about that icon you mentioned you might design on your own?

Thanks!
